server/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -620,11 +629,7 @@ public IndexSettings getIndexSettings() {
     }
 
     public String getType() {
-        return mapperService.documentMapper() == null ? null : mapperService.documentMapper().type();
-    }
-
-    public boolean typeExists(String type) {
-        return mapperService.documentMapper(type) != null;
```

Comment:
If there is a default mapping, this call `mapperService.documentMapper(type)` returns a non-null mapper for the `_default_` type. So this refactor may change the `type` query behavior slightly. This seems acceptable since default mappings were deprecated in 6.0 and the `type` query isn't completely clear on this behavior, but it felt worth pointing out.

Reply:
I think it only returns something for `_default_` if there isn't another mapping. Which, I think, only happens when the index is in the process of being made. Though, I can fiddle with it a bit to see if a totally empty mapping gets it.

Reply:
I played around a bit - I don't think you can have documents in the index and not have a `DocumentMapper`. So I *think* this is safe.

Reply:
And if it isn't then that change is quite small.